### PR TITLE
basic head of execution_manager

### PR DESCRIPTION
**Pull request by jackzihaozhou**
Basic implementation of starting a container in server and copy exe file to /tmp inside docker container. Job Manager can directly call "ExecutionManager().submit_task(task)" to submit a task to Execution Manager.

### CODE REVIEW

File: felucca/backend/execution_manager.py [hunk 1]
```diff
@@ -1,2 +1,143 @@
...
+         
+        parser.add_argument('-ooanalyzer', dest="ooanalyzer",action = "store_true",default=False)
+        parser.add_argument('-j','--json', dest="j", type = str)
+        parser.add_argument('-R','--prolog-results ', dest="R", type = str)
+        parser.add_argument('-n','--new-method', dest="n", type = str)
+        parser.add_argument('-F','--prolog-facts', dest="F", type = str)
+        parser.add_argument('-f','--file', dest="f", type = str)
+        #command_line_input = "ooanalyzer --json output.json -F facts -R results --file oo.exe"
+        #print(command_line_input)
+        if(command_line_input[0]!='-'):
```
Comment (Orankarl) [inline felucca/backend/execution_manager.py:29]: Looks like Java style.

File: felucca/backend/execution_manager.py [hunk 2]
```diff
@@ -1,2 +1,143 @@
...
+import tarfile
+import docker
+from common.singleton import Singleton
+from threading import Thread
+
+@Singleton
 class ExecutionManager(object):
-    pass
\ No newline at end of file
+
```
Comment (Orankarl) [inline felucca/backend/execution_manager.py:13]: Add comments in class & methods as description.
Also add some comments for code blocks as explanation.

File: felucca/backend/execution_manager.py [hunk 3]
```diff
@@ -1,2 +1,143 @@
...
+        
+        new_command_line_input = ""
+        
+        for key in dict:
+            if type(dict[key])==bool:
+                new_command_line_input = new_command_line_input+key+" "
+            elif dict[key] is not None:
+                new_command_line_input = new_command_line_input+"-"+key+" "+dict[key]+" "
+                
+
```
Comment (Orankarl) [inline felucca/backend/execution_manager.py:60]: Extra empty lines.

File: felucca/backend/execution_manager.py [hunk 4]
```diff
@@ -1,2 +1,143 @@
...
+    def copy_to_container(self,src,dst,container):
+
+
+        os.chdir(os.path.dirname(src))
+        srcname = os.path.basename(src)
+
+        tar = tarfile.open(src + '.tar', mode='w')
+        try:
+            tar.add(srcname)
+        finally:
```
Comment (xkbt777) [inline felucca/backend/execution_manager.py:76]: try without except?

File: felucca/backend/execution_manager.py [hunk 5]
```diff
@@ -1,2 +1,143 @@
...
+from common.singleton import Singleton
+from threading import Thread
+
+@Singleton
 class ExecutionManager(object):
-    pass
\ No newline at end of file
+    
+    def __init__(self):
+        self.map_id2Task_container = {}
```
Comment (xkbt777) [inline felucca/backend/execution_manager.py:15]: id2Task looks like camel case

File: felucca/backend/execution_manager.py [hunk 6]
```diff
@@ -1,2 +1,143 @@
...
+        #self.map_id_Task[task.task_ids] = task
+        
+        self.map_id2Task_container[task.task_ids] = (task,container)
+        return(True);
+        
+    def run_container_flask(self,container):
+        print(f"running: 'flask run --host=0.0.0.0' inside container shell")
+
+        #exec_log = container.exec_run("flask run --host=0.0.0.0" ,stdout=True,stderr=True,stream=True)
+        exec_log = container.exec_run("echo \" 666  \" " ,stdout=True,stderr=True,stream=True)
```
Comment (xkbt777) [inline felucca/backend/execution_manager.py:96]: why use "echo 666"?

File: felucca/backend/execution_manager.py (hunk 2 above)
Comment (jackzihaozhou) [reply felucca/backend/execution_manager.py:13]: get it

File: felucca/backend/execution_manager.py (hunk 6 above)
Comment (jackzihaozhou) [reply felucca/backend/execution_manager.py]: Just for testing

**Comment by moody6520** [comment]
integrate with other pr
https://github.com/MITS-Felucca/Felucca/pull/120

**Pull request closed without merging**